src/Pate/Memory/MemTrace.hs

```diff
@@ -619,10 +619,13 @@ data MemTraceImpl sym ptrW = MemTraceImpl
   }
 
 data MemTraceState sym ptrW = MemTraceState
-  { memArr :: MemTraceArr sym ptrW }
+  { memArrBytes :: MemTraceArrBytes sym ptrW
+  , memArrRegions :: MemTraceArrRegions sym ptrW
+  }
 
 type MemTraceSeq sym ptrW = SymSequence sym (MemEvent sym ptrW)
-type MemTraceArr sym ptrW = MemArrBase sym ptrW (BaseBVType 8)
+type MemTraceArrBytes sym ptrW = MemArrBase sym ptrW (BaseBVType 8)
+type MemTraceArrRegions sym ptrW = MemArrBase sym ptrW BaseIntegerType
 
 type MemArrBase sym ptrW tp = RegValue sym (SymbolicArrayType (EmptyCtx ::> BaseIntegerType) (BaseArrayType (EmptyCtx ::> BaseBVType ptrW) tp))
```

Comment:
Can you add a short description of how memory is arranged based on these types?  I think the basic structure can be worked out from the types, but a description would be nonetheless helpful.

More importantly, can you record the policy for choosing the region for a written pointer value?